torchvision/prototype/transforms/_utils.py

```diff
@@ -55,13 +55,30 @@ def _get_defaultdict(default: T) -> Dict[Any, T]:
     return defaultdict(functools.partial(_default_arg, default))
 
 
-def _setup_fill_arg(fill: Union[FillType, Dict[Type, FillType]]) -> Dict[Type, FillType]:
+def _convert_fill_arg(fill: features.FillType) -> features.FillTypeJIT:
+    # Fill = 0 is not equivalent to None, https://github.com/pytorch/vision/issues/6517
+    # So, we can't reassign fill to 0
+    # if fill is None:
+    #     fill = 0
+    if fill is None:
+        return fill
+
+    # This cast does Sequence -> List[float] to please mypy and torch.jit.script
+    if not isinstance(fill, (int, float)):
+        fill = [float(v) for v in list(fill)]
+    return fill
+
+
+def _setup_fill_arg(fill: Union[FillType, Dict[Type, FillType]]) -> Dict[Type, FillTypeJIT]:
     _check_fill_arg(fill)
 
     if isinstance(fill, dict):
-        return fill
+        fill_copy = {}
+        for k, v in fill.items():
+            fill_copy[k] = _convert_fill_arg(v)
+        return fill_copy
```

Comment:
Let's convert the input dict/value into a `Dict[Type, FillTypeJIT]` dictionary on the constructor of objects. This way we won't have to do this on the runtime ever time. This is less of a speed optimization and more of code-quality as it allows us to remove a lot of dup code.